R/icu_los.R

```diff
@@ -50,10 +50,16 @@
 #' derived numeric fields "icu_los_hrs_derived" and "icu_los_days_derived".
 #'
 #' @note:
-#' By design, function will not return any NA values.
-#' Encounter IDs in the `cohort` table that are not present in the `ipscu` table are assumed to have no visits to ICU.
-#' For these encounters, a value of 0 will be assigned to the derived fields.
-#' When one tries to left-join the output of this function to another table, make sure the list of encounters aligns in both tables
+#' Encounter IDs in the `cohort` table that are not present in the `ipscu` table
+#' are assumed to have no visits to ICU. For these encounters, a value of 0 will
+#' be assigned to the derived fields.
+#' Encounter IDs in the `ipscu` table that have any missing/invalid
+#' `scu_admit_date_time` or `scu_discharge_date_time` will be returned with
+#' `icu_los = NA`. Some of those entries have valid date information (but no
+#' timestamp). Users may choose to impute missing timestamps prior to running
```

Comment:
Well explained. And I think it would be better to add the missing timestamp sentence to the warning in case the user does not read this documentation carefully, they can at least notice it in the warning, as this is currently a issue affecting a whole site and this site is included in most projects.

Reply:
Good point, I updated the warning message to `"Identified a total of XXX genc_ids with at least 1 missing/invalid 'scu_admit_date_time' or 'scu_discharge_date_time'. These entries will be returned as 'icu_los = NA'.
 Some of these entries might have valid date information (but no timestamp). Please carefully check the 'ipscu' table and consider whether it might be useful to impute missing timestamps."` Let me know if this is better.